server/src/main/java/org/elasticsearch/index/query/DistanceFeatureQueryBuilder.java

```diff
@@ -108,7 +108,7 @@ public String getWriteableName() {
 
     @Override
     protected Query doToQuery(QueryShardContext context) throws IOException {
-        MappedFieldType fieldType = context.fieldMapper(field);
+        MappedFieldType fieldType = context.getFieldType(field);
```

Comment:
I find some of these usages suspicious: do they really mean to check if the returned field type is null after the unmapped fields handling (unmapped fields may get mapped as text), or would they rather need to call `isFieldMapped` ?

Reply:
I'm pretty sure these should be calling isFieldMapped.  We should maybe think about moving the unmapped field handling out of QueryShardContext and have it at the callsites that really need it? It's a bit too magic otherwise.